src/ui/wxWidgets/PasswordSafeFrame.cpp

```diff
@@ -890,21 +890,35 @@ void PasswordSafeFrame::RefreshToolbarButtons()
   if (toolbar->GetToolCount() > 0) {
     toolbar->ClearTools();
   }
+  bool toolbarShowText = pref->GetPref(PWSprefs::ToolbarShowText);
+  bool showMenuSeprator = pref->GetPref(PWSprefs::ShowMenuSeparator);
 
   for (const auto & PwsToolbarButton : PwsToolbarButtons) {
-    if (pref->GetPref(PWSprefs::ShowMenuSeparator) && (PwsToolbarButton.IsSeparator())) {
-      toolbar->AddSeparator()->SetId(PwsToolbarButton.id);
+    if (PwsToolbarButton.IsSeparator()) {
+      if (showMenuSeprator)
+        toolbar->AddSeparator()->SetId(PwsToolbarButton.id);
     }
     else {
-      toolbar->AddTool(
-        PwsToolbarButton.id,
-        wxGetTranslation(PwsToolbarButton.toollabel),
-        PwsToolbarButton.GetBitmapForEnabledButton(),
-        PwsToolbarButton.GetBitmapForDisabledButton(),
-        wxITEM_NORMAL,
-        wxGetTranslation(PwsToolbarButton.tooltip),
-        wxEmptyString,
-        nullptr);
+      if(toolbarShowText)
+        toolbar->AddTool(
+          PwsToolbarButton.id,
+          wxGetTranslation(PwsToolbarButton.toollabel),
+          PwsToolbarButton.GetBitmapForEnabledButton(),
+          PwsToolbarButton.GetBitmapForDisabledButton(),
+          wxITEM_NORMAL,
+          wxGetTranslation(PwsToolbarButton.tooltip),
+          wxEmptyString,
+          nullptr);
+      else
+        toolbar->AddTool(
+          PwsToolbarButton.id,
+          wxGetTranslation(PwsToolbarButton.toollabel),
+          PwsToolbarButton.GetBitmapForEnabledButton(),
+          PwsToolbarButton.GetBitmapForDisabledButton(),
+          wxITEM_NORMAL,
+          wxEmptyString,
```

Comment:
This is the only line that's different, why not use a ternary operator here?
```suggestion
          toolbarShowText ? wxGetTranslation(PwsToolbarButton.tooltip) : wxEmptyString,
```

Reply:
Because compiler error about ambiguous return value of wxTranslate.

Reply:
If someone doesn't know that this is a hotfix and stumbles upon it, then I wouldn't be surprised if they change that again as part of a code clean-up. Smells like code duplication.  8^)